### ISSUE TITLE
Métadonnées - scénario pour chapitre de mémoire ou de thèse - création du schéma

### ISSUE THREAD

**Issue by RochDLY** [EcrituresNumeriques/stylo#1024]
Ce ticket comporte les éléments nécessaires à la création du scénario de métadonnées pour un chapitre de mémoire ou de thèse.

**Les métadonnées sont présentées comme suit : Nom courant (nom technique)(type)(obligatoire)**

J'ai découpé les métadonnées en "groupes" pour plus de lisiblité et créé des objets pour éviter la redondances dans le ticket (par exemple, un traducteur ou un auteur appelle l'objet 'individu" qui contient les champs noms, prénoms, etc.).

1. **Au niveau du corpus**

Objet individu:

- Nom (surname)(str)
- Prénom (forename) (attention, actuellement la clé est forname, voir le ticket [#1018](https://github.com/EcrituresNumeriques/stylo/issues/1018))(str)
- ORCID (orcid)(str, exemple : '0000-0002-3519-4365')
- Biographie (biography)(str, plutôt long, entre 500 et 1000 signes)
- Affiliations (affiliations)(str, doit accepter plusieurs entrées, exemple : 'Université de Montréal' 'CRCEN' 'CRIHN')
- Email (email)(str)
- VIAL (viaf)(str)
- FOAF (foaf)(str)
- Wikidata (wikidata)(str)
- ISNI (isni)(str)
- Fonction(academicTitle)(str)
- Role(role)(str)

---

Groupe des personnes : 

- Auteur.ice.s (authors)(individu)
- Directeur.ice.s de recherche (Directors)(individu)
- Membres du jury(juryMembers)(individu)

---

Groupe de données de la thèse :

?(thesis:)

- Titre de la thèse (title_f)(str)
- Discipline (discipline)(str, laisser la possibilité d'en mettre plusieurs)
- Titre traduit (translatedTitle:)
    - langue (lang)(select language)
    - titre (text_f)(str)
- Langue
- Mot-clés contrôlés de la thèse
- Mot-clés de la thèse
    - langue
    - texte
- Nombre de pages de la thèse
- Sous-titre de la thèse
- URL de la thèse (urlBook)(str)
- Date de publication (date)(calendar)
- Date de soutenance(defenseDate)(calendar)
- Résumé de la thèse (abstract:)
    - langue (lang)(select language)
    - texte (text_f)(str, plutôt un texte long entre 500 et 1000 caractères)

---

Groupe de données sur l'établissement :

- Établissement(s) d'attache: NB il faut en mettre plusieurs
        -  Nom de l'université (universityName)(str) 
        - Département (department)(str)
        -  Faculté (faculty)(str) 
        - École doctorale(ecoleDoc)(str)

2. **Au niveau de l'article**

Groupe des données du chapitre : 

- Identifiant du chapitre (id)(str)
- Titre du chapitre (title_f)(str)(obligatoire)
- Sous-titre du chapitre (subtitle_f)(str)
- URL du chapitre (urlChapter)(str)
- Nombre de pages du chapitre (nbPage)(à calculer ?)
- Mot-clés contrôlés du chapitre (controlledKeywords:)(select keywords)
    - idRameau: ''
    - label: ''
    - uriRameau: ''
- Mot-clés du chapitre (keywords:)
    - langue (lang)(select language)
    - texte (list_f:)(list)

**Comment by ggrossetie** [comment]
On a un concept de "localized content" avec : 

- lang
- title
- subtitle
- abstract
- keywords


De mon point de vue ça peut remplacer : 

- Titre de la thèse (title_f)(str)
- Titre traduit (translatedTitle:)
    - langue (lang)(select language)
    - titre (text_f)(str)
- Langue
- Mot-clés de la thèse
    - langue
    - texte
- Sous-titre de la thèse
- Résumé de la thèse (abstract:)
    - langue (lang)(select language)
    - texte (text_f)(str, plutôt un texte long entre 500 et 1000 caractères)


On aura à la racine les champs ci-dessous pour la langue principale de la thèse : 

- lang
- title
- subtitle
- abstract
- keywords

Puis une liste de "localized content" pour fournir le titre, le sous-titre, le résumé et les mot-clés dans une autre langue.

Concernant `affiliations` on a déjà cette propriété dans les métadonnées des articles avec un type `string`. Est-ce qu'on modifie pour passer sur un type `array` (avec des éléments `string`) ?
Même question pour `discipline` où on indique "laisser la possibilité d'en mettre plusieurs.

Si on utilise un type `string`, on peut renseigner plusieurs affiliations ou discipline mais les données ne seront pas structurées car le séparateur est "libre". Si on utilise un type `array` il sera possible, dans les templates, d'itérer sur la liste des valeurs.

**Comment by ggrossetie** [comment]
Est-ce qu'on ajoute les propriétés ci-dessous à l'objet générique "personne" : 

- Fonction(academicTitle)(str)
- Role(role)(str)

Ou est-ce que ces informations sont uniquement nécessaires pour certains "types de personnes" ?

**Comment by gromettoclara** [comment]
> Est-ce qu'on ajoute les propriétés ci-dessous à l'objet générique "personne" :
> 
> * Fonction(academicTitle)(str)
> * Role(role)(str)
> 
> Ou est-ce que ces informations sont uniquement nécessaires pour certains "types de personnes" ?

ça me paraît plus simple de les mettre partout en les rendant non obligatoire, sauf si vous pensez que ça alourdit inutilement le formulaire. Pour l'auteur c'est effectivement pas hyper pertinent... Quoi que... Pour un HDR par exemple tu as déjà une fonction...

**Comment by gromettoclara** [comment]
![Image](https://github.com/user-attachments/assets/b64349ab-e3c5-4261-a867-9474c20a326a)

des title à rajouter dans le JSON ?

**Comment by ggrossetie** [comment]
Oui, il faut modifier le fichier https://github.com/EcrituresNumeriques/stylo/blob/master/front/src/schemas/corpus-thesis-ui-schema.json afin de transformer le bloc `items` dans `affiliatedInstitutions` : 

https://github.com/EcrituresNumeriques/stylo/blob/96b246e8b61b8d870eb1fe57c50eb3aec2e443e0/front/src/schemas/corpus-thesis-ui-schema.json#L294C4-L301


Cela doit ressembler à :  

```json
{
  "affiliatedInstitutions":{
    "ui:add-item-title":"affiliatedInstitutions.itemAdd",
    "ui:className":"affiliatedInstitutions",
    "items":{
      "universityName":{
        "ui:title":"universityName"
      },
      "department":{
        "ui:title":"department"
      },
      "faculty":{
        "ui:title":"faculty"
      },
      "doctoralSchool":{
        "ui:title":"doctoralSchool"
      }
    }
  }
}
```

Et ensuite il faut rajouter ces clés dans les fichiers de traductions `form.json`. Par exemple : 


`front/src/locales/en/form.json`

```json
{
  "universityName": "University name"
}
```

**Comment by ggrossetie** [comment]
@gromettoclara j'ai détaillé ce qu'il y avait à faire au cas où tu souhaites faire le correctif mais si tu n'a pas le temps/envie je peux m'en occuper 😉

**Comment by gromettoclara** [comment]
je prends !! demain lundi ou mardi
